### ISSUE TITLE
[Feature] Write a documentation for it

### ISSUE THREAD

**Issue by fnabinash** [fnabinash/rust-practice#16]
Create documentation for all exercises with context.

Also include commands, helpful suggestions and tutorials to make it easy to understand for everyone.

**Comment by unpeturbed** [comment]
Hey. Are you saying we should add documentation to every exercise?

**Comment by fnabinash** [comment]
I think we should create documentation for the guide.

Every exercise has some little content about it like "What the exercise about?", "What output might look like?", "Links to related resources", and if in future lots of people solve these exercises then we will feature top solutions of every exercise at bottom of the page.

Also, there is a separate page dedicated to the roadmap and top resources to learn Rust.

For now, I have these ideas in my mind. If you have something else, please let me know.

**Comment by fnabinash** [comment]
Alright. I just released the first version of the practice branch. If you have some free time you can give some feedback on it.

**Comment by unpeturbed** [comment]
Okay. I think i get you now. I am not supposed to send a pull request to merge my codes to practice. i'm supposed to just solve and leave it at that. RIght?

**Comment by fnabinash** [comment]
I have some different plans.

- We will create documentation for this guide but not as the comments in the `src/main.rs` files. (Because it makes more cultured so it may confuse newcomers.)
- I think we should create documentation separate(a special folder of markdown files) and link them to exercises. This means, the `src/main.rs` file contains the problem statement at the top and then a link to the documentation file(i.e. to the markdown file in another folder like we have created a separate folder for `images`), then the main functions.
- In future, we will make this a binary create and some tests to it but for now I think we should not include tests.

Please share your thoughts on this. Do not think this is my personal project. This is for the community for everyone, so feel free to share your thoughts. Also, mention ideas or feedback you think if we implement it will be more helpful to the community.

**Comment by unpeturbed** [comment]
This is good. I also think we can include the binary crates (lib.rs) for tests from the intermediary section.

**Comment by fnabinash** [comment]
Why not, It is a good idea.

**Comment by unpeturbed** [comment]
Yeah. Writing tests is part of development and i believe it's great to start early so it becomes part of us. It's good practice
